doc/charm++/manual.rst

```diff
@@ -3001,6 +3001,11 @@ The detailed APIs of these two methods are described as follows:
    after AtSync() and before ResumeFromSync(), so they can make some
    assumptions about state).
 
+   .. note:: Dynamic insertion works with periodic load balancing with
+	     no issues. However, dynamic deletion does not, since
+	     deletion may occur while the load balancing strategy is
```

Comment:
Is there any way to support this from the user-end? As in, could I do `X` before/after deletion? I ask because with AMR we would want to delete the old object that we split into multiple.

Reply:
Currently, I think not. I've been trying some quick hacky things to get it to work, but to no avail, fixing this is going to require changes to the core location/migration code. The fundamental problem here is that deletions that occur while LB is running (which there is no way of preventing in periodic mode because LB and the application are completely decoupled) can cause the balancer to try to migrate an object that no longer exists, so it'll segfault if we try to send it anyway, or the send/receive counts won't match if we don't send it, causing a hang.

There are of course ways around this, namely empty messages for deleted objects, buffering object deletions while in LB, or something else of that ilk.

In general, we expect that "real applications" will use periodic load balancing mode only sparingly, so fixing this hasn't really been a priority for us. Would this be something that'd be particularly useful for you all?

Reply:
:+1: I figured it would be rather difficult to deal with on the bookkeeping end. Thanks for explaining that! One user-facing improvement would be if there's a way to avoid a segfault and instead print an error message. I realize this might not be possible because of the amount of type erasure needed inside Charm++ and how the type erasure is done :)

At this point I wouldn't feel comfortable requesting periodic LB with deletion :) I think we first need to get all the LBs working and behaving as we expect for some interesting physics problems before we can make the call that the `AtSync` synchronization is a bottleneck. A possibly more useful/relevant feature would be to allow `AtSync` to occur with an explicit QD (or reduction) from the user. I could see us being able to take advantage of the sync by doing a reduction to compute a new optimal time step size or something else like that. However, even that I wouldn't say is worth putting any effort into right now

Reply:
Sorry, I wasn't clear, it currently does print an error like:
```
------------- Processor 3 Exiting: Called CmiAbort ------------
Reason: [3] LBDB::Migrate: Handle 13 no longer registered, range 0-25
```
I meant that it would cause a segfault if we added that feature without overhauling the migration code.

Okay, gotcha. Also, to be clear, do you meant that you'd like to trigger `AtSync` via QD or reduction? Doing so (on a single proxy) is pretty simple right now if you just create an entry method that just contains  `{ AtSync(); }`, and that can be passed as a broadcast callback to either QD or a reduction. Doing so across several proxies is a bit more complicated, but also possible via a similar mechanism. Is there something I'm missing about your request, or is it just a desire to get this functionality with more streamlined syntax?

Reply:
Ah, okay, perfect!

Yep, you understood correctly. I didn't know that would work already. Thanks! I wouldn't worry about a more streamlined syntax or anything until we have figured out what actually works well in practice :)

Reply:
Great, makes sense. I think it'd be a nice little spoonful of syntactic sugar, so thanks for the suggestion.

Also, I've created #3270 to track the deletion of objects during LB.